jit-compiler/src/codegen.rs

```diff
@@ -332,30 +361,38 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
                 let (vars, code) = check_pattern(var_name, pattern)?;
                 // TODO if we want to explicitly specify the type, we need to exchange the non-captured
                 // parts by `()`.
-                format!("let {vars} = (|{var_name}{ty}| {code})({value}).unwrap();",)
+                (
+                    format!("let {vars} = (|{var_name}{ty}| {code})({value}).unwrap();"),
+                    var_height + new_vars,
+                )
+            }
+            StatementInsideBlock::Expression(e) => {
+                (format!("{};", self.format_expr(e, var_height)?), var_height)
             }
-            StatementInsideBlock::Expression(e) => format!("{};", self.format_expr(e)?),
         })
     }
 
     /// Returns a string expression evaluating to the value of the symbol.
     /// This is either the escaped name of the symbol or a deref operator
     /// applied to it.
-    fn symbol_reference(&self, symbol: &str) -> String {
-        let needs_deref = if is_builtin::<T>(symbol) {
-            false
+    fn symbol_reference(&self, symbol: &str, type_args: &[Type]) -> String {
+        let type_args = if type_args.is_empty() {
+            "".to_string()
         } else {
-            let (_, def) = self.analyzed.definitions.get(symbol).as_ref().unwrap();
-            if let Some(FunctionValueDefinition::Expression(typed_expr)) = def {
-                !matches!(typed_expr.e, Expression::LambdaExpression(..))
+            format!("::<{}>", type_args.iter().map(map_type).join(", "))
+        };
+        if is_builtin::<T>(symbol) {
+            return format!("Callable::Fn({}{type_args})", escape_symbol(symbol));
+        }
+        let (_, def) = self.analyzed.definitions.get(symbol).as_ref().unwrap();
+        if let Some(FunctionValueDefinition::Expression(typed_expr)) = def {
+            if matches!(typed_expr.e, Expression::LambdaExpression(..)) {
+                format!("Callable::Fn({}{type_args})", escape_symbol(symbol))
```

Comment:
are there tests using type_args?

Reply:
Yes, the last test calls `std::convert::fe`, which is a generic function.